bodo/libs/_memory.cpp

```diff
@@ -1404,15 +1466,24 @@ std::tuple<bool, int64_t, int64_t, int64_t> BufferPool::get_alloc_details(
 
 void BufferPool::free_helper(uint8_t* ptr, bool is_mmap_alloc,
                              int64_t size_class_idx, int64_t frame_idx,
-                             int64_t size_aligned) {
+                             int64_t size_aligned, int64_t alignment) {
     bool frame_pinned;
     if (is_mmap_alloc) {
         frame_pinned =
             this->size_classes_[size_class_idx]->isFramePinned(frame_idx);
         this->size_classes_[size_class_idx]->FreeFrame(frame_idx);
     } else {
         frame_pinned = true;
+
+#if defined(_WIN32)
+        if (alignment < kMinAlignment) {
```

Comment:
I assume this is the inverse of `alignment > kMinAlignment` in `Allocate`.
```suggestion
        if (alignment <= kMinAlignment) {
```